### ISSUE TITLE
Variable-width text fields processed as constant-width ones

### ISSUE THREAD

**Issue by luthir** [M0r13n/pyais#94]
What am I trying to do:
`print(*pyais.encode_msg(pyais.messages.MessageType12.create(mmsi=111111111, dest_mmsi=999999999, text='TEST')), sep='\n')
`
What I expect to get:
`!AIVDM,1,1,,A,<1aucikfJjOtD5CD,0*55
`
What I actually get
```
!AIVDM,3,1,0,A,<1aucikfJjOtD5CD000000000000000000000000000000000000000000000,0*57
!AIVDM,3,2,0,A,0000000000000000000000000000000000000000000000000000000000000,0*27
!AIVDM,3,3,0,A,0000000000000000000000000000000000000000000000,0*16
```

As stated in https://gpsd.gitlab.io/gpsd/AIVDM.htm width of "text" field in MessageType12 and MessageType14 may be shorter than given maximum width, but pyais treats these fields as a constat-width ones, and trails given short text with a bunch of @ signs up to maximum length.

Seems it would be better to somehow calculate actual required width of the field while encoding

**Comment by M0r13n** [comment]
Yes, you are totally right. I will look into this.

**Comment by M0r13n** [comment]
Hey @luthir,

I patched this bug. I added a new keyword to the `bit_field`: `variable_length`. This is a boolean value that is `False` by default and that can be used to tell the encoder whether or not to add trailing padding (`@`) to text/data fields.

This affects messages of type **6, 8, 12, 17**.

The output for this example program now produces the following output:

**Driver program**:
```py
from pyais.encode import encode_msg
from pyais.messages import MessageType6, MessageType12

msg = MessageType6.create(mmsi='366053209', dest_mmsi='1111', data=b'TEST')
encoded = encode_msg(msg)
print(encoded)

msg = MessageType12.create(mmsi='366053209', dest_mmsi='1111', text='TEST')
encoded = encode_msg(msg)
print(encoded)
```

**Result**:

```txt
['!AIVDO,1,1,,A,65M67F@0015L001DAE=D,0*2D']
['!AIVDO,1,1,,A,<5M67F@0015LD5CD,0*59']
```

Before this fix, the same driver code produced the following output for type 12:

`['!AIVDO,3,1,0,A,<5M67F@0015LD5CD00000000000000000000000000000000000000000000,0*6B', '!AIVDO,3,2,0,A,000000000000000000000000000000000000000000000000000000000000,0*15', '!AIVDO,3,3,0,A,000000000000000000000000000000000000000000000000,0*14']`

This fix will be available shortly.

**Comment by M0r13n** [comment]
Can be installed with version 2.3.1

**Comment by luthir** [comment]
Great work, thank you!
But, as far as I can see, there already was no problem with MessageType6 and MessageType8, since variable length field data has been as-is-encoded bytes type, not padded text. Also, instead of fixing MessageType14, MessageType17 was modified.
Please consider rework possibility

**Comment by M0r13n** [comment]
> as far as I can see, there already was no problem with MessageType6 and MessageType8, since variable length field data has been as-is-encoded bytes type, not padded text

Technically, you are correct. Those messages were already correctly encoded. But because they are variable in length, I decided to set the flag `variable_length=True` even tough it doesn't change anything. True to the motto: "*explicit is better than implicit*".

> Also, instead of fixing MessageType14, MessageType17 was modified.

For type 17 the same applies as for type 6 and 8. I wanted to make the intention explicit. The data field is variable in length. But I definitely forgot type 14. Nice catch!

**Comment by luthir** [comment]
Here is also a fix to broken test #98

**Comment by M0r13n** [comment]
Thanks again! 🥇
